lib/Driver/ToolChains/CommonArgs.cpp

```diff
@@ -870,6 +870,21 @@ tools::ParsePICArgs(const ToolChain &ToolChain, const ArgList &Args) {
     PIP = true;
   }
 
+  // Pagerando and '-ffunction-sections'/'-fdata-sections' are incompatible
+  bool UseSeparateSections = isUseSeparateSections(Triple);
+  bool FunctionSections = Args.hasFlag(options::OPT_ffunction_sections,
+                                       options::OPT_fno_function_sections,
+                                       UseSeparateSections);
+  bool DataSection = Args.hasFlag(options::OPT_fdata_sections,
+                                  options::OPT_fno_data_sections,
+                                  UseSeparateSections);
+  if (PIP && (FunctionSections || DataSection))
```

Comment:
If we want to allow mixing, I would forfeit the checks here here and simply revert to the old behavior: the backend silently ignores `-ffunction-sections` for pagerando functions.

The comment should probably go to where the main `-fpip`/`-fpagerando` flag is defined.